doc/iterations/2023.6/shapes/filterTreeBasedRepresentations.adoc

```diff
@@ -0,0 +1,86 @@
+= (M) Shape
+
+== Problem
+
+It is quite difficult to search for/identify specific tree item(s) in a tree when this tree contains lots of tree items.
+
+== Key Result
+
+The solution should not slow-down the user interface. The user interface must remains as smooth as it was before the developments of this filter.
+The results displayed by the filter should be visible immediately, i.e. without any latency.
+Otherwise, if the solution can only be implemented with a latency, then during the computation time, a "busy" indicator should be displayed, informing the user his request is processing.
+
+== Solution
+
+Add a search capacity in the Explorer View and other tree based representations, by providing a textual filtering of the tree just like what is being done in VSCode using a simple textfield.
+
+=== Breadboarding
+
+A shortcut (most likely ctrl+f) will be available in the Explorer View, allowing to display a search bar under the toolbar at the top of the Explorer View.
+This search bar will be composed by a simple textfield, a filter button and a close button.
+When user type a value in the textfield of the search bar, the tree item(s) containing this value are highlighted (only the part corresponding to the value, not the whole tree item(s)).
+Each character added/removed in the search bar will update the result (i.e. the highlighted tree item(s)).
+The filter button allows to hide all visible leaf tree items that do not match the search.
+The close button close the search bar, the highlighted results are not highlighted anymore, and the filtered tree items are not filtered anymore.
+The up/down arrow keys allow to jump from an highlighted item to the next one/previous one.
+
+=== Scenario 1 - Search
+
+User want to search tree items containing a specific value as their label or part of their label.
+User type the shortcut to enable the search bar.
+User type the value to search in the search bar.
+The tree items containing the searched value are highlighted (only the part of the items containing the value).
+Each character added/removed in the search bar updates the highlighted tree items.
+User close the search bar with the close button.
+All highlighted items are not highlighted anymore.
+
+=== Scenario 2 - Filter
+
+User want to filter tree items containing a specific value as their label or part of their label.
+User type the shortcut to enable the search bar.
+User type the value to search in the search bar.
+The tree items containing the searched value are highlighted (only the part of the items containing the value).
```

Comment:
How do we represent that there are some matches available under a collapsed item?

Reply:
Yes, if it is possible it should be part of the shape.

Reply:
I don't see how the frontend could know that "there are some matches available under a collapsed item" so I don't see how that could be done. VS Code does not do it either because it's a complex question even for them.